Automattic/jetpack
# Jetpack plugin: Integrate Stats package

This PR integrates the Stats package into the Jetpack plugin. The stats package (docs TBA in a follow up PR) is now the source of truth in terms of:
- Managing stats related options (see `Automattic\Jetpack\Stats\Options`)
- Building the tracking pixel data and adding it to the footer ( see `Automattic\Jetpack\Stats\Tracking_Pixel`)
- Registering the stats related XMLRPC endpoints (see `Automattic\Jetpack\Stats\XMLRPC_Provider`)
- Communicating with the WPCOM REST API Stats endpoints (see `Automattic\Jetpack\Stats\WPCOM_Stats`)

#### Changes proposed in this Pull Request:

**Adding the Stats package dependency**
* Updated the Jetpack plugin's `composer.json` with the Stats package dependency
* Updated the config package accordingly

* Managing stats options is now handled by `Automattic\Jetpack\Stats\Options`
**Stats options**
* deprecated `stats_get_options`
* deprecated `stats_get_option`
* deprecated `stats_set_options`
* deprecated `stats_set_option`
* deprecated `stats_upgrade_options`

**Tracking pixel**
* Building the tracking pixel data and adding it to the footer is now handled by `Automattic\Jetpack\Stats\Tracking_Pixel`
* removed `stats_array`
* deprecated `stats_add_to_footer`
* deprecated `stats_render_footer`
* deprecated `stats_render_amp_footer`
* deprecated `stats_build_view_data`

**XMLRPC endpoints**
* Stats related XMLRPC endpoints are now handled by `Automattic\Jetpack\Stats\XMLRPC`
* removed `stats_xmlrpc_methods`
* deprecated `stats_get_blog`
* removed `stats_esc_html_deep`

**Note:** Replacing the `stats_get_from_restapi` function in favor of `WPCOM_Stats` will be done in a follow-up PR as we are not deprecating that function yet and prefer not to overburden this one.

#### Other information:


#### Jetpack product discussion


p9dueE-5Rd-p2

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






- Check the suggested test cases by the test-case-reminder bot
- Play around with the Jetpack plugin's stat settings and make sure they are respected
<img width="1041" alt="Screenshot 2022-10-06 at 09 56 14" src="https://user-images.githubusercontent.com/1758399/194234811-d2a7d1af-643f-4ef2-a096-19f8b5dd0f62.png">

- Make sure we track pageviews with [Infinite Scroll enabled](https://jetpack.com/support/infinite-scroll/)
- Make sure we track [Carousel](https://jetpack.com/support/carousel/) views 
- Pay extra attention to the tracking data we send and make sure the jp version is present as before (`j` param). As you might note the stats package is not longer adding this to the default data we are tracking, as its plugin agnostic. The Jetpack plugin is adding this parameter now via the `stats_array` filter
- Make sure to compare all the different places we show stats related data with `trunk` with the Stats module enabled AND disabled (you can disable the module by visiting `wp-admin/admin.php?page=jetpack_modules`):
- Site Stats in Jetpack dashboard
 <img width="1072" alt="Screenshot 2022-10-06 at 09 43 30" src="https://user-images.githubusercontent.com/1758399/194233275-92280f9c-cc70-43fa-a489-d768418dd918.png">   

- Site Stats page in Jetpack plugin
<img width="1109" alt="Screenshot 2022-10-06 at 09 46 21" src="https://user-images.githubusercontent.com/1758399/194233268-d29756dc-3594-4976-9104-9b8b230dff95.png">

- Stats widget in wp-admin home
<img width="604" alt="Screenshot 2022-10-06 at 09 47 25" src="https://user-images.githubusercontent.com/1758399/194233261-76ede159-4d57-47d8-a136-bd683cc4a4d1.png">
